core/kazoo_documents/src/kzd_blacklists.erl

```diff
@@ -46,6 +294,99 @@ numbers(Doc, Default) ->
 set_numbers(Doc, Numbers) ->
     kz_json:set_value([<<"numbers">>], Numbers, Doc).
 
+-spec format_numbers(kz_term:api_object()) -> kz_term:api_object().
+format_numbers('undefined') ->
+    'undefined';
+format_numbers(Doc) ->
+    case is_blacklist(Doc) of
+        'true' ->
+            Numbers = kz_json:get_value(<<"numbers">>, Doc),
+            format_numbers(Numbers);
+        'false' -> kz_json:map(fun format_number_map/2, Doc)
+    end.
+
+-spec format_number_map(kz_term:ne_binary(), kz_json:object()) ->
+                               {kz_term:ne_binary(), kz_json:object()}.
+format_number_map(Number, Data) ->
+    {knm_converters:normalize(Number), Data}.
+
+-spec numbers_name(doc()) -> kz_term:api_binary().
+numbers_name(Doc) ->
+    numbers_name(Doc, 'undefined').
+
+-spec numbers_name(doc(), Default) -> binary() | Default.
+numbers_name(Doc, Default) ->
+    kz_json:get_binary_value([<<"numbers">>, <<"name">>], Doc, Default).
+
+-spec set_numbers_name(doc(), binary()) -> doc().
+set_numbers_name(Doc, NumbersName) ->
+    kz_json:set_value([<<"numbers">>, <<"name">>], NumbersName, Doc).
+
+-spec owner_id(doc()) -> kz_term:api_ne_binary().
+owner_id(Doc) ->
+    owner_id(Doc, 'undefined').
+
+-spec owner_id(doc(), Default) -> kz_term:ne_binary() | Default.
+owner_id(Doc, Default) ->
+    kz_json:get_ne_binary_value([<<"owner_id">>], Doc, Default).
+
+-spec set_owner_id(doc(), kz_term:ne_binary()) -> doc().
```

Comment:
Contract disagrees with the implementation; looks like `api_ne_binary()` is allowed.